### PR TITLE
Add tests for multi cluster handling

### PR DESCRIPTION
**Pull request by bjosv**
Verifies that multiple clusters can be accessed simultaneously using the regular or the async API.

### CODE REVIEW

File: tests/ct_connection.c [hunk 1]
```diff
@@ -66,22 +69,87 @@ void test_password_missing() {
...
-    redisClusterAsyncDisconnect(cc);
+    ExpectedResult *expect = (ExpectedResult *)privdata;
+    if (expect->noreply) {
+        assert(reply == NULL);
+        assert(strcmp(cc->errstr, expect->errstr) == 0);
+    } else {
+        assert(reply != NULL);
+        assert(reply->type == expect->type);
+        if (reply->type != REDIS_REPLY_INTEGER)
+            assert(strcmp(reply->str, expect->str) == 0);
```
Comment (zuiderkwast) [inline tests/ct_connection.c:149]: Not integer doesn't imply it's a string reply, does it? Can't it be an array or anything else...?

File: tests/ct_connection.c [hunk 2]
```diff
@@ -134,9 +204,10 @@ void test_async_password_wrong() {
     assert(acc->cc->err == REDIS_ERR_OTHER);
     assert(strncmp(acc->cc->errstr, "WRONGPASS", 6) == 0);
 
-    // Test connection
-    status = redisClusterAsyncCommand(acc, commandCallback, (char *)"THE_ID",
-                                      "SET key1 Hello");
+    // No connection
+    ExpectedResult r;
+    status =
+        redisClusterAsyncCommand(acc, commandCallback, &r, "SET key1 Hello");
```
Comment (zuiderkwast) [inline tests/ct_connection.c:210]: Perhaps a matter of personal taste, but I suggest we keep the old layout with the linebreak before `"SET key1 Hello"`. Or is this clang-format's preferred style?

File: tests/ct_connection.c [hunk 3]
```diff
@@ -168,8 +239,9 @@ void test_async_password_missing() {
     assert(strncmp(acc->cc->errstr, "NOAUTH", 6) == 0);
 
     // No connection
-    status = redisClusterAsyncCommand(acc, commandCallback, (char *)"THE_ID",
-                                      "SET key1 Hello");
+    ExpectedResult r;
+    status =
+        redisClusterAsyncCommand(acc, commandCallback, &r, "SET key1 Hello");
```
Comment (zuiderkwast) [inline tests/ct_connection.c:244]: See above.

**Review by zuiderkwast — COMMENTED** [review]
Nice! Only a few minor comments on style.

File: tests/ct_connection.c (hunk 1 above)
Comment (bjosv) [reply tests/ct_connection.c:149]: Yes, this looked a bit awkward now. Changed it to only check for types that uses the `str` element in the reply

File: tests/ct_connection.c (hunk 2 above)
Comment (bjosv) [reply tests/ct_connection.c:210]: Yes, clang-format is changing it to the worse.. maybe we should think about if we can live without it..
I changed a bit so it looks ok again..

**Review by zuiderkwast — APPROVED** [review]
So you changed `result` to `ret` to avoid some linebreaks?

OK, but we can also blindly follow clang format and put an end to such discussions. :-) I just wasn't sure it was clang format's idea to change this.

**Pull request merged**